lib/plugins/aws/invokeLocal/index.js

```diff
@@ -373,45 +404,70 @@ class AwsInvokeLocal {
-        this.checkDockerDaemonStatus(),
-        this.checkDockerImage().then(exists => {
-          return exists ? {} : this.pullDockerImage();
-        }),
-        this.getLayerPaths().then(layerPaths => this.buildDockerImage(layerPaths)),
-        this.extractArtifact(),
-      ]).then(
-        results =>
-          new BbPromise((resolve, reject) => {
-            const imageName = results[2];
-            const artifactPath = results[3];
-            const configuredEnvVars = this.getConfiguredEnvVars();
-            const envVarsFromOptions = this.getEnvVarsFromOptions();
-            const envVars = _.merge(configuredEnvVars, envVarsFromOptions);
-            const envVarsDockerArgs = _.flatMap(envVars, (value, key) => [
-              '--env',
-              `${key}=${value}`,
-            ]);
-            const dockerArgsFromOptions = this.getDockerArgsFromOptions();
-            const dockerArgs = _.concat(
-              ['run', '--rm', '-v', `${artifactPath}:/var/task`],
-              envVarsDockerArgs,
-              dockerArgsFromOptions,
-              [imageName, handler, JSON.stringify(this.options.data)]
-            );
-            const docker = spawn('docker', dockerArgs);
-            docker.stdout.on('data', buf => this.serverless.cli.consoleLog(buf.toString()));
-            docker.stderr.on('data', buf => this.serverless.cli.consoleLog(buf.toString()));
-            docker.on('exit', error => {
-              return error ? reject(error) : resolve(imageName);
+    return this.ensurePackage()
+      .then(() => this.checkDockerDaemonStatus())
+      .then(() =>
+        BbPromise.all([
+          this.checkDockerImage(`lambci/lambda:${runtime}`).then(exists => {
+            return exists ? {} : this.pullDockerImage();
+          }),
+          this.getLayerPaths().then(layerPaths => this.buildDockerImage(layerPaths)),
+          this.extractArtifact(),
+        ]).then(results => {
+          const imageName = results[1];
+          const artifactPath = results[2];
+          const configuredEnvVars = this.getConfiguredEnvVars();
+          const envVarsFromOptions = this.getEnvVarsFromOptions();
+          const envVars = _.merge(configuredEnvVars, envVarsFromOptions);
+          const envVarsDockerArgs = _.flatMap(envVars, (value, key) => [
+            '--env',
+            `${key}=${value}`,
+          ]);
+          const dockerArgsFromOptions = this.getDockerArgsFromOptions();
+          const dockerArgs = _.concat(
+            ['run', '--rm', '-v', `${artifactPath}:/var/task`],
+            envVarsDockerArgs,
+            dockerArgsFromOptions,
+            [imageName, handler, JSON.stringify(this.options.data)]
+          );
+
+          return spawnExt('docker', dockerArgs)
+            .then(({ stdoutBuffer, stderrBuffer }) => {
+              this.printBuffers(stderrBuffer, stdoutBuffer);
+              return imageName;
+            })
+            .catch(({ code, stdoutBuffer, stderrBuffer }) => {
```

Comment:
This should be second callback to then (otherwise we're also handling here errors from first `then` callback)